ironfish-cli/src/commands/accounts/repair.ts

```diff
@@ -0,0 +1,276 @@
+
+  static description = `Repairs wallet database stores`
+
+  static flags = {
+    ...LocalFlags,
+  }
+
+  static args = [
+    {
+      name: 'account',
+      parse: (input: string): Promise<string> => Promise.resolve(input.trim()),
+      required: false,
+      description: 'Name of the account to repair the database for',
+    },
+  ]
+
+  async start(): Promise<void> {
+    const { args } = await this.parse(Repair)
+
+    const node = await this.sdk.node()
+    await NodeUtils.waitForOpen(node)
+
+    const account = this.loadAccount(node.wallet, args.account)
+
+    this.log(`Repairing wallet for account ${account.name}`)
+
+    this.log('Repairing decryptedNotes and balance')
+    await this.repairDecryptedNotes(account, node.wallet.walletDb, node.chain)
+
+    this.log('Repairing nullifierToNote')
+    await this.repairNullifierToNoteHash(account, node.wallet.walletDb)
+
+    this.log('Repairing sequenceToNoteHash')
+    await this.repairSequenceToNoteHash(account, node.wallet.walletDb, node.chain)
+  }
+
+  private loadAccount(wallet: Wallet, accountName: string | undefined): Account {
+    if (accountName) {
+      const account = wallet.getAccountByName(accountName)
+      if (account) {
+        return account
+      }
+      throw new Error(`No account with name ${accountName}`)
+    }
+
+    const defaultAccount = wallet.getDefaultAccount()
+    if (defaultAccount) {
+      return defaultAccount
+    }
+
+    throw new Error('Could not find an account to repair.')
+  }
+
+  private async repairDecryptedNotes(
+    account: Account,
+    walletDb: WalletDB,
+    chain: Blockchain,
+  ): Promise<void> {
+    let unconfirmedBalance = 0n
+
+    let noteUnspentMismatches = 0
+    let nullifierNoteHashMismatches = 0
+
+    await walletDb.db.transaction(async (tx) => {
+      for await (const decryptedNoteValue of account.getNotes()) {
+        const transactionValue = await account.getTransaction(
+          decryptedNoteValue.transactionHash,
+          tx,
+        )
+
+        Assert.isNotUndefined(
+          transactionValue,
+          'Account has a note but is missing the transaction that it received the note from. Account must be rescanned using `accounts:rescan --reset`',
+        )
+
+        await this.verifyBlockContainsTransaction(
+          decryptedNoteValue.transactionHash,
+          transactionValue.blockHash,
+          chain,
+        )
+
+        await walletDb.setNoteHashSequence(
+          account,
+          decryptedNoteValue.hash,
+          transactionValue.sequence,
+          tx,
+        )
+
+        if (!decryptedNoteValue.nullifier) {
+          if (transactionValue.sequence) {
+            throw new Error(
+              'Transaction marked as on chain, but note missing nullifier. Account must be rescanned using `accounts:rescan --reset`.',
+            )
+          }
+
+          continue
+        }
+
+        const spent = await chain.nullifiers.contains(decryptedNoteValue.nullifier)
```

Comment:
Here I think we may want to do `chain.getHashAtSequence` instead since we were having issues with the nullifier contains. This would depend on running the other repair command correctly

Reply:
Discussed on a call -- we still need to check `chain.nullifiers` because we don't have a way to determine the transaction or the block that a note was spent on.